torch/_tensor.py

```diff
@@ -1026,6 +1027,61 @@ def __torch_function__(cls, func, types, args=(), kwargs=None):
             else:
                 return _convert(ret, cls)
 
+    def __dlpack__(self, stream=None):
+        """
+        Creates a dlpack capsule of the current tensor to
+        be exported to other libraries.
+
+        This function will be called from the `from_dlpack` method
+        of the consumer library.
+
+        Args:
+            stream (integer or None): A Python integer representing a pointer
+            to a stream (CUDA or ROCm). `stream` is provided by the consumer
+            to the producer to instruct the producer to ensure that operations
```

Comment:
How can the "consumer" provide the stream? Doesn't the user provide the stream?

Reply:
`__dlpack__` is called from `from_dlpack` method and it will pass the current stream as an argument as part of the dlpack specification. Will update the docstring to reflect this!